libraries/helpers.rb

```diff
@@ -0,0 +1,57 @@
+#
+# Cookbook: consul
+# License: Apache 2.0
+#
+# Copyright (C) 2015 Bloomberg Finance L.P.
+#
+
+module ConsulCookbook
+  module Helpers
+    extend self
+
+    def arch_64?
+      node['kernel']['machine'] =~ /x86_64/ ? true : false
+    end
+
+    def windows?
+      node['os'].eql?('windows') ? true : false
+    end
+
+    def program_files
+      'C:\\Program Files' + (arch_64? ? '' : ' x(86)')
+    end
+
+    def default_environment
+      {
+        'GOMAXPROCS' => [node['cpu']['total'], 2].max.to_s,
+        'PATH' => '/usr/local/bin:/usr/bin:/bin'
+      }
+    end
+
+    def command(config_file, config_dir)
```

Comment:
I created this file to try and keep everything DRY. Some methods are only used once (e.g. default_environment) but others are used in several places (e.g. windows?, program_files). I prefer to keep provider & resource classes as clean as possible.